cobalt/browser/h5vcc_experiments/h5vcc_experiments_impl.cc

```diff
@@ -0,0 +1,156 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/browser/h5vcc_experiments/h5vcc_experiments_impl.h"
+
+#include "base/feature_list.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/metrics/field_trial_params.h"
+#include "build/build_config.h"
+#include "cobalt/browser/cobalt_experiment_names.h"
+#include "cobalt/browser/global_features.h"
+#include "components/prefs/pref_service.h"
+
+namespace h5vcc_experiments {
+
+namespace {
+
+mojom::OverrideState GetFeatureInternal(
+    std::map<std::string, h5vcc_experiments::mojom::OverrideState>&
+        override_map,
+    const std::string& feature_name) {
+  // auto accessor = cobalt::GlobalFeatures::GetInstance()->accessor();
+  // base::FeatureList::OverrideState override =
+  //     accessor->GetOverrideStateByFeatureName(
+  //         feature_name);
+  // if (override == base::FeatureList::OverrideState::OVERRIDE_ENABLE_FEATURE)
+  // {
+  //   return h5vcc_experiments::mojom::OverrideState::OVERRIDE_ENABLE_FEATURE;
+  // } else if (override ==
+  // base::FeatureList::OverrideState::OVERRIDE_DISABLE_FEATURE) {
+  //   return h5vcc_experiments::mojom::OverrideState::OVERRIDE_DISABLE_FEATURE;
+  // }
+  // return h5vcc_experiments::mojom::OverrideState::OVERRIDE_USE_DEFAULT;
+
+  // Cache feature params in a map for faster lookup.
+  if (override_map.empty()) {
+    std::string enable_features;
+    std::string disable_features;
+
+    base::FeatureList::GetInstance()->GetFeatureOverrides(&enable_features,
+                                                          &disable_features);
+
+    std::string feature;
+    std::string study;
+    std::string group;
+    std::string feature_params;
+
+    for (const auto& enabled :
+         base::FeatureList::SplitFeatureListString(enable_features)) {
+      if (!base::FeatureList::ParseEnableFeatureString(
+              enabled, &feature, &study, &group, &feature_params)) {
+        // TODO(b/407734134): Register UMA for non-parsable features.
+        LOG(ERROR) << "Unable to parse enabled feature: " << enabled;
+        continue;
+      }
+      if (!feature.empty() && feature[0] == '*') {
+        override_map[feature] =
+            h5vcc_experiments::mojom::OverrideState::OVERRIDE_USE_DEFAULT;
+      }
+      override_map[feature] =
+          h5vcc_experiments::mojom::OverrideState::OVERRIDE_ENABLE_FEATURE;
+    }
+    for (const auto& disabled :
```

Comment:
Because you have the same loop here twice with the only difference being the use of `enable_features` and `disable_features` I suspect this can be made into a function and simplified. As-is seems a bit verbose. This assumes we even need this cache at all.